lib/src/repository/devices/devices_repository.dart

```diff
@@ -195,8 +201,9 @@ class Device {
 abstract class Trait {
   late final String name;
   late final State state;
+  late final List<Property> properties;
 
-  Trait(this.name, this.state);
+  Trait(this.name, this.state, this.properties);
```

Comment:
@surajadh @rlperez This is a discussion point if we want to continue with this approach and using wrapper classes. 

From what I read, it's good practice to build wrapper classes around generated code instead of using generated code directly. I would vote to continue using these wrapper classes. If we continue with them, maybe we can move them to a separate file away from `devices_repository.dart`

Reply:
Yes I agree. This part of code needs some cleanup. I do not have concrete implementation in mind yet but this needs to be organized in such a way that attributes are properly encapsulated within its bounding trait. I do not think we need a generic sort of attribute floating around beyond its trait.